azerothcore/azerothcore-wotlk
# fix(Core/Creatures): Added new AI function `OnTeleportUnreacheablePlayer` to teleport all unreachable players.

Fixed #11750



## Changes Proposed:
-  Creature with this flag will teleport unreachable (due to mmaps) current target to itself.
-  Non-raid mobs that cannot reach (due to mmaps) its current target will switch to another target (if any) instead of evading 

## Issues Addressed:

- Closes #11750

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

`.go creature id 11382`
Aggro boss
`.gm fly on`
.go xyz -12191.968750 -1947.059570 152`
Wait at least 5 sec to see that Mandokir teleports you


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
